ensemble-view/src/main/java/hec/ensembleview/charts/PanMouseAdapter.java

```diff
@@ -0,0 +1,257 @@
+
+        //make an assumption that the pointer component is a Viewport. May need to make a pan component tool interface??
+        if (pointerComponent instanceof Viewport) {
+            Viewport v = (Viewport) pointerComponent;
+            boolean repaint = false;
+            if (dxl != 0) {
+                panAxis(v.getAxis("x1"), dxl);
+                panAxis(v.getAxis("x2"), dxl);
+                repaint = true;
+            }
+
+            if (dyl != 0) {
+                panAxis(v.getAxis("y1"), dyl);
+                panAxis(v.getAxis("y2"), dyl);
+                repaint = true;
+            }
+            if (repaint) {
+                previousMouseDragged = currentPoint;
+                v.getG2dPanel().repaint();
+            }
+        }
+    }
+
+    /**
+     * pan an axis by a given amount "deltaA" in local coordinates.
+     * @param a
+     * @param dal
+     */
+    private void panAxis(Axis a, double dal) {
+
+        if (a == null || a.getZoom() == 1) {
+            return;
+        }
+        double vMaxw = a.getViewMax();
+        double vMinw = a.getViewMin();
+
+        double vMaxl = a.w2l(vMaxw);
+        double vMinl = a.w2l(vMinw);
+
+        double nMaxl = vMaxl + dal;  //new view maximum in local (pixel) coords
+        double nMinl = vMinl + dal;  //new view minimum in local (pixel) coords
+
+        double nMaxw = a.l2w((int) nMaxl); //new max world
+        double nMinw = a.l2w((int) nMinl); //new min world
+
+        if (nMaxw > a.getMax()) {
+            //view range local
+            double viewRangeLocal = Math.abs(vMaxl - vMinl);
+            nMaxw = a.getMax();
+            nMaxl = a.w2l(a.getMax());
+            nMinw = a.l2w((int) (a.isReversed() ? nMaxl + viewRangeLocal : (nMaxl - viewRangeLocal)));
+        }
+        if (vMinw < a.getMin()) {
+            double viewRangeLocal = vMaxl - vMinl;
+            nMinl = a.w2l(a.getMin());
+            nMaxw = a.l2w((int) (nMinl + viewRangeLocal));
+            nMinw = a.l2w((int) (a.isReversed() ? (nMinl - viewRangeLocal) : (nMinl + viewRangeLocal)));
+        }
+        a.setViewLimits(nMinw, nMaxw);
+    }
+
+    @Override
+    public void mouseWheelMoved(java.awt.event.MouseWheelEvent e) {
+        if (e.getScrollAmount() != 0) {
+            int direction;
+
+            JScrollBar toScroll;
+            if (e.isShiftDown()) {
+                toScroll = null;   //horizontal Scroll bar;
+            } else if (e.isControlDown()) {
+                toScroll = null;   //vertical Scroll bar;
+            } else {
+                mouseWheelZoom(e);
+                return;
+            }
+
+            direction = e.getWheelRotation() < 0 ? -1 : 1;
+            if (e.getScrollType() == java.awt.event.MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+                scrollByUnits(toScroll, direction, e.getScrollAmount());
```

Comment:
Holding CTRL or Shift down and moving the mouse wheel throws NPEs since the `toScroll` parameter being passed here always seems to be null.

Reply:
same as above comment